lib/ide/src/executor/web.rs

```diff
@@ -0,0 +1,104 @@
+//! Module defining `JsExecutor` - an executor that tries running until stalled
+//! on each animation frame callback call.
+
+use crate::prelude::*;
+
+use basegl::control::callback::CallbackHandle;
+use basegl::control::EventLoopCallback;
+use basegl::control::EventLoop;
+use futures::task::LocalSpawn;
+use futures::task::LocalFutureObj;
+use futures::task::SpawnError;
+use futures::executor::LocalPool;
+use futures::executor::LocalSpawner;
+
+/// Executor. Uses a single-threaded `LocalPool` underneath, relying on basegl's
+/// `EventLoop` to do as much progress as possible on every animation frame.
+#[derive(Debug)]
+pub struct JSExecutor {
+    /// Underlying executor. Shared internally with the event loop callback.
+    executor    : Rc<RefCell<LocalPool>>,
+    /// Executor's spawner handle.
+    pub spawner : LocalSpawner,
+    /// Event loop that calls us on each frame.
+    event_loop  : Option<EventLoop>,
+    /// Handle to the callback - if dropped, loop would have stopped calling us.
+    /// Also owns a shared handle to the `executor`.
+    cb_handle   : Option<CallbackHandle>,
+}
+
+impl JSExecutor {
+    /// Creates a new JS Executor. It is not yet running, use `schedule_running`
+    /// method to schedule it in an event loop.
+    pub fn new() -> JSExecutor {
+        let executor  = LocalPool::default();
+        let spawner   = executor.spawner();
+        let executor  = Rc::new(RefCell::new(executor));
+        JSExecutor {
+            executor,
+            spawner,
+            event_loop : None,
+            cb_handle  : None,
+        }
```

Comment:
if you create `let event_loop = None;` and `let cb_handle = None` then you can write this in a single line, which will make the whole code 4-lines shorter.

Reply:
The code is longer and more symbols are repeated.
But if you really want… 
done